test/apps/data-filter/src/scatterplot-layer.js

```diff
@@ -0,0 +1,38 @@
+import {ScatterplotLayer} from '@deck.gl/layers';
+import dataFilter from './data-filter';
+
+const defaultProps = {
+  getFilterValue: 1
+};
+
+export default class ScatterplotLayerWithFilter extends ScatterplotLayer {
+
+  getShaders() {
+    const shaderSettings = super.getShaders();
+
+    shaderSettings.modules.push(dataFilter);
+
+    shaderSettings.inject = {
+      'vs:#decl': `
+attribute float instanceFilterValue;
+`,
+      'vs:#main-end': `
+filter_setVisibility(instanceFilterValue);
+`,
+      'fs:#main-end': `
+gl_FragColor = filter_filterColor(gl_FragColor);
+`
+    };
+    return shaderSettings;
+  }
+
+  initializeState() {
+    super.initializeState();
+
+    this.getAttributeManager().addInstanced({
```

Comment:
I wonder if the module description could capture that some values are typically new attributes is some general way?

```
export default {
  name: 'dataFilter'
  attributes: {
    filterValue: GL.FLOAT,
  },
};
```

Reply:
How does this simplify usage?